src/main/java/org/inferred/freebuilder/processor/ToStringGenerator.java

```diff
@@ -25,6 +27,18 @@ public static void addToString(
       Datatype datatype,
       Map<Property, PropertyCodeGenerator> generatorsByProperty,
       boolean forPartial) {
+    generatorsByProperty = generatorsByProperty.entrySet().stream()
+        .filter(e -> e.getKey().isInToString())
+        .collect(
+            Collectors.toMap(
+                Map.Entry::getKey,
+                Map.Entry::getValue,
+                (u, v) -> {
+                  throw new IllegalStateException(String.format("Duplicate key %s", u));
```

Comment:
No need for this: since you got the stream from a Map, this code cannot be hit. Use the two-parameter form of `toMap`.

Reply:
The test code is relying on a deterministic ordering of the map entries. So to maintain that contract, I'm constructing a `LinkedHashMap`. Unfortunately the jdk doesn't expose their `throwingMerger` and I'm not aware of the jdk providing a `toLinkedHashMap()` out of the box.

Reply:
Ah! And keeping order is also a "principal of least surprise" contract. Carry on then 😄 

Could you comment that the boilerplate is there to keep entries in order? Then anyone refactoring it will know what the constraint is.